### ISSUE TITLE
Loading spinner -> skeleton loading

### ISSUE THREAD

**Issue by terezanovotna** [patternfly/patternfly-design#726]
This is an enhancement for an existing pattern. 

Pattenfly Spinner could be substituted with skeleton loading which looks much cleaner on the screen.  Or skeleton loading could be introduced as a new pattern to pick from when any information is being loaded. 

![47005198-e9c3d080-d13b-11e8-96b6-78b06d9fc004](https://user-images.githubusercontent.com/36040135/47015214-acb40a00-d14c-11e8-8d1d-b5fa895f16e4.gif)

![46404134-9ba6da00-c704-11e8-9c40-13d242578a3d](https://user-images.githubusercontent.com/36040135/47015200-a2920b80-d14c-11e8-9ad6-b4953a3e1a94.gif)

The idea originated in https://github.com/theforeman/foreman/pull/6103#issuecomment-430162419

**Comment by mcarrano** [comment]
Can you say more about this @terezanovotna ? Not sure exactly what the request is. Is it to incrementally load form data as it becomes available vs. waiting for all the data? Or is it a different kind of spinner element?

**Comment by terezanovotna** [comment]
It is a suggestion/idea on using different visual to load data instead of a spinner. There is no high priority on this, but as the devs suggested it, I thought it could be a good idea to implement it in the future.

**Comment by katierik** [comment]
Following up on this, I have a proposed design for how to use skeleton loaders & spinners together. This is going to be used in a product currently in development. One of the devs on the project (Ryan L) has been working on a "smart skeleton" which could be appropriate 

Google share: https://docs.google.com/presentation/d/1kfKbl5HtumYiNm-ubJxxP93EqJjfThijEd6n1QIhD60/edit?usp=sharing 
PDF copy of slides - message me

**Comment by mcarrano** [comment]
Fantastic work @katierik . I'd love to get this on the PF4 roadmap somewhere. @rachael-phillips what do you think?

**Comment by rachael-phillips** [comment]
Yes, absolutely. Skeleton loading is currently in our backlog, but if we decide this is something that we  will work on the next few months we should definitely consider the skeleton loading designs that @katierik has put together! Spinner has been under design and development with PF4 over the past couple of days, so that component will be coming soon.

**Comment by ninavizz** [comment]
Hi @katierik! I tried opening/viewing your skeleton pattern, and got an "Access Denied" message. Is that intentional?

Looking forward to seeing a good Skeleton pattern added to Patternfly! I just discovered Patternfly, and am so delighted to see Redhat making this contribution to design. 😄

**Comment by andybraren** [comment]
+1’ing this potential new component. OpenShift has been working on some new dashboard designs, and we’d use this new pattern extensively while data is being loaded. Here’s an example of one card’s loading state:

![health-1-0](https://user-images.githubusercontent.com/9122899/60441866-e39c5400-9be5-11e9-8466-4b11002cc2b2.png)

@matthewcarleton put together [this HTML/CSS example](https://codepen.io/matthewcarleton/pen/xedQqK?editors=1100), and the [Insights Storybook](https://cloud.redhat.com/docs/storybook?path=//story/components--skeleton) seems to have an even more refined React implementation that would be perfect for us too. I imagine the circular variant in the image above (for an undetermined status icon) could probably be done with a CSS class modifier on the same component.

**Comment by rachael-phillips** [comment]
@mcarrano should this be moved to the patternfly-next repo or it is ok for it to stay here for now?

**Comment by mcarrano** [comment]
@rachael-phillips Yes, this should be moved to the -next repo so we can track it.
